onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -285,6 +315,23 @@ void ModelBuilder::RegisterModelInputs() {
         case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
           type = Type::TENSOR_FLOAT32;
           break;
+        case ONNX_NAMESPACE::TensorProto_DataType_UINT8: {
+          // For ONNX the quantized input does not carry scale and zero point info
+          // So we will need to search the operator using this input
+          // And dig out the scale and zero point as the input initializers to the operator
+          type = Type::TENSOR_QUANT8_ASYMM;
+          if (!Contains(all_quantized_op_inputs, input_name)) {
+            // We current do not support uint8 input if it is not a quantized input
+            ORT_THROW("The input of graph doesn't have valid type, name: " +
+                      input_name + " type: " +
+                      std::to_string(type_proto->tensor_type().elem_type()));
+          }
```

Comment:
Consider refactoring to return a status. Cheaper in terms of binary size if exceptions are enabled, and more graceful if exceptions are disabled.

Reply:
Added a TODO for this, Task 825192: [NNAPI EP] Change throw in ModelBuilder to return status

---
In reply to: [456963094](https://github.com/microsoft/onnxruntime/pull/4530#discussion_r456963094) [](ancestors = 456963094)